addons/account/models/account_tax.py

```diff
@@ -1061,7 +1061,12 @@ def grouping_key_generator(base_line, tax_values):
         for grouping_key, tax_values in global_tax_details['tax_details'].items():
             if tax_values['currency_id']:
                 currency = self.env['res.currency'].browse(tax_values['currency_id'])
-                res['totals'][currency]['amount_tax'] += currency.round(tax_values['tax_amount'])
+                tax_amount = tax_values['tax_amount']
+                line = tax_values['records'] and next(iter(tax_values['records']))
+                company = line and line.company_id or self.company_id
```

Comment:
Don't infer the company from the record. Add the company as an extra parameter instead.
Edit: retrieve the company from the tax?

Reply:
It seems ok to retrieve it from the tax, thanks for the hint